src/GameLogic/DataStructures/ActionSequence.ts

```diff
@@ -0,0 +1,24 @@
+import {PlayerAction} from "../Enums";
+
+export class ActionSequence {
+    private _actionSequence: PlayerAction[];
+    private _step: number = 0;
+
+	constructor(actionSequence: PlayerAction[] = [])
+	{
+		this._actionSequence = actionSequence;
```

Comment:
Since you'll be modifying the array, make a copy of it so that it doesn't accidentally modify the one that was passed.

Alternatively, if you don't want to make an eager copy store a flag that will determine whether you have a copy yet or not, and make the copy when you want to modify it for the first time.